server/src/main/java/org/elasticsearch/action/admin/indices/rollover/MaxAgeCondition.java

```diff
@@ -57,7 +57,12 @@ public String getWriteableName() {
 
     @Override
     public void writeTo(StreamOutput out) throws IOException {
-        //TODO here we should just use TimeValue#writeTo and same for de-serialization in the constructor, we lose information this way
+        // While we technically could serialize this would out.writeTimeValue(...), that would
+        // doing the song and dance around backwards compatibility for this value. Since in this
```

Comment:
"that would doing the song and dance" -> "that would require doing the song and dance"

Reply:
:+1: good catch